File: Core/Shared/JavaLib/src/net/lukx/jchatter/lib/models/CFile.java
package net.lukx.jchatter.lib.models;

import net.lukx.jchatter.lib.PublicApi;

import java.util.Date;

/***
 * Class used for deserialization of server data
 */
@SuppressWarnings("unused")
@PublicApi
public class CFile {
    /***
     * UUID of file
     */
    public byte[] uuid;
    /***
     * ID of the user who uploaded this file.
     * FK to User's ID
     */
    public int iduploader;
    /***
     * ID of the room this file has been uploaded to.
     * FK to Room's ID
     */
    public int idroom;
    /***
     * Name of this file
     */
    public String fileName;
    /***
     * When this file was uploaded
     */
    public Date dateUploaded;
    /***
     * If this file has expired and it's contents
     * are no longer on the server
     */
    public boolean expired;
}

File: Core/Wpf/ChatterWpf/ChatterWpf/Models/CFile.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ChatterWpf.Models
{
    public class CFile
    {

        public byte[] uuid{get;set;}

        public int iduploader{get;set;}

        public int idroom{get;set;}

        public string fileName{get;set;}

        public DateTime dateUploaded{get;set;}

        public bool expired{get;set;}
    }

}